### PR TITLE
Extend types accepted by DatasetContext to include (SolidDataset & WithAcp)

### PR DESCRIPTION
**Pull request by ameyabhat**
This commit allows the DatasetContext component to be used with (SolidDataset & WithAcp)

# Checklist

- [x] All acceptance criteria are met.
- [n/a] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [n/a] New functions/types have been exported in `index.ts`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by Vinnl — COMMENTED** [review]
@ABhat00 Could you clarify what error you're running into? Anything that accepts a `SolidDataset` should also accept a `(SolidDataset & WithAcp)` as far as I'm aware (and a `(SolidDataset & WithResourceInfo)`, for that matter).

**Comment by ABhat00** [comment]
Right now, the DatasetContext component will accept a `(SolidDataset & WithACP)` but will only return a `SolidDataset`, which means that I'm not able to use any of the acp functions on the dataset I've kept in context. I was looking to make the component return the`(SolidDataset & WithAcp)` if that was passed to it to solve that issue

**Comment by Vinnl** [comment]
@ABhat00 Ah I see - that requires using a [generic](https://www.typescriptlang.org/docs/handbook/generics.html) to link the input type to the output type. See for example how we've done that for [`setThing`](https://github.com/inrupt/solid-client-js/blob/137ba17e8655b20667dcf9fed5c591225588655a/src/thing/thing.ts#L166-L169), whose output SolidDataset has the same type as the input, with the addition of `WithChangeLog`. (There, `Dataset` is the generic.)

Also note that your `import` statement imports from the `dist` folder, rather than directly from `@inrupt/solid-client`. Although Webpack 4 and below allow just reaching into a package and importing from whatever file is available there, this will break once this library migrates to Webpack 5.

File: src/context/datasetContext/index.tsx [hunk 1]
```diff
@@ -27,9 +27,10 @@ import {
 } from "@inrupt/solid-client";
 
 import useDataset from "../../hooks/useDataset";
+import { WithAcp } from "@inrupt/solid-client/dist/acp/acp";
```
Comment (ThisIsMissEm) [inline src/context/datasetContext/index.tsx:30]: This needs to be changed to the following, afaik:
```suggestion
import { WithAcp } from "@inrupt/solid-client/acp/acp";
```

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]

**Comment by NSeydoux** [comment]
This library is being sunset.

**Pull request closed without merging**